Assets/SEE/UI/PropertyDialog/UserPropertyDialog.cs

```diff
@@ -0,0 +1,162 @@
+            networkConfig.Load();
+
+            dialog = new GameObject("User settings");
+            PropertyGroup group = dialog.AddComponent<PropertyGroup>();
+            group.Name = "User settings";
+            {
+                playerName = dialog.AddComponent<StringProperty>();
+                playerName.Name = "Username";
+                playerName.Value = networkConfig.Playername;
+                playerName.Description = "Username which will be used as avatar-tag and in other function.";
+                group.AddProperty(playerName);
+            }
+            // Dialog
+            propertyDialog = dialog.AddComponent<PropertyDialog>();
+            propertyDialog.Title = "User settings";
+            propertyDialog.Description = "Enter the user settings";
+            propertyDialog.AddGroup(group);
+
+            // Register listeners
+            propertyDialog.OnConfirm.AddListener(OKButtonPressed);
+            propertyDialog.OnCancel.AddListener(CancelButtonPressed);
+
+            SEEInput.KeyboardShortcutsEnabled = false;
+            // Go online
+            propertyDialog.DialogShouldBeShown = true;
+        }
+
+        /// <summary>
+        /// Notifies all listeners on <see cref="OnCancel"/> and closes the dialog.
+        /// </summary>
+        private void CancelButtonPressed()
+        {
+            propertyDialog.Close();
+            OnCancel.Invoke();
+            SEEInput.KeyboardShortcutsEnabled = true;
+            Close();
+        }
+
+        /// <summary>
+        /// Notifies all listeners on <see cref="OnConfirm"/> and save name and closes the dialog.
+        /// </summary>
+        private void OKButtonPressed()
+        {
+            bool errorOccurred = false;
+            {
+                string playername = playerName.Value.Trim();
+
+                if (ValidUsername(playername))
+                {
+                    networkConfig.Playername = playername;
+                }
+                else
+                {
+                    ShowNotification.Error
+                       ("Username Error",
+                        "Username needs to be minimum one character long.");
+                    errorOccurred = true;
+                }
+            }
+            if (!errorOccurred)
+            {
+                propertyDialog.Close();
+                OnConfirm.Invoke();
+                SEEInput.KeyboardShortcutsEnabled = true;
+                Close();
+                callBack?.Invoke();
+            }
+        }
+
+        /// <summary>
+        /// Checks if playername is at least one character long. Other restrictions could be added in this method too.
+        /// </summary>
+        /// <param name="playername">the playername to validate</param>
+        /// <returns>true if <paramref name="playername"/> conforms the restrictions</returns>
+        private static bool ValidUsername(string playername)
+        {
+            return !String.IsNullOrWhiteSpace(playername);
```

Comment:
Should use primitive instead of boxed type:
```suggestion
            return !string.IsNullOrWhiteSpace(playername);
```